knimin/templates/pm_target_gene_library_prep.html

```diff
@@ -0,0 +1,281 @@
+      var fields = {'plates': JSON.stringify(plates), 'robot': robot, 'tm300': tm300,
+                    'tm50': tm50, 'master_mix': masterMix, 'water': water};
+      $.each(fields, function(key, val) {
+        $('<input>').attr({type: "hidden", name: key, value: val}).appendTo(form);
+      });
+      form.appendTo('body').submit();
+    }
+  }
+
+  /**
+   *
+   * Checks if the current value of the element exists in the given array
+   * and updated its style accordingly
+   *
+   * @param {Element} elem The input DOM element
+   * @param {Array} existing The array of existing elements
+   *
+   **/
+  function formatInput(elem, existing) {
+    var val = $(elem).val().trim();
+    $(elem).removeClass('pm-warning');
+    if (val.length > 0 && $.inArray(val, existing) === -1) {
+      $(elem).addClass('pm-warning');
+    }
+  };
+
+  /**
+   *
+   * Adds the list of plates to the given select element
+   *
+   * @param {Element} elem The select DOM element
+   *
+   **/
+  function addDNAPlateOptions(elem) {
+    var plates = {% raw dumps(dna_plates) %};
+    $.each(plates, function (idx, plate) {
+      var display = "(ID: " + plate.id + ") " + plate.name + " - " + plate.date;
+      $('<option>').attr('value', plate.id).text(display).appendTo(elem);
+    });
+  };
+
+  /**
+   *
+   * Add the list of barcode plates to the given select element
+   *
+   * @param {Element} elem The select DOM element
+   *
+   **/
+  function addBarcodePlateOptions(elem) {
+    var plates = {% raw dumps(primer_plates) %};
+    $.each(plates, function (idx, plate) {
+      var display = plate.name + " - " + plate.target_gene_region;
+      $('<option>').attr('value', plate.id).text(display).appendTo(elem);
+    });
+  };
+
+  $(document).ready(function() {
+    var masterMixLots = {% raw dumps([v['name'] for v in mastermixlots]) %};
+    var waterLots = {% raw dumps([v['name'] for v in waterlots]) %};
+
+    // Initialize the master mix lot autocompletion
+    $(".autocomplete-master-mix").autocomplete({source: masterMixLots,
+                                                select: function (e, ui){
+                                                  formatInput(this, masterMixLots)
+                                              }});
+    $(".autocomplete-master-mix").keyup(function(e) {
+      formatInput(this, masterMixLots);
+    });
+
+    // Initialize the water lot autocompletion
+    $(".autocomplete-water").autocomplete({source: waterLots,
+                                           select: function (e, ui){
+                                             formatInput(this, waterLots)
+                                           }});
+    $(".autocomplete-water").keyup(function(e) {
+      formatInput(this, waterLots);
+    });
+
+    addDNAPlateOptions($('#init-plate-select'));
+    addBarcodePlateOptions($('#init-primer-select'));
+    $('.plate-select').select2();
+    $('.glyphicon-warning-sign').hide();
```

Comment:
It is possible that on the page there is more than one, but when loading the page (this is getting executed at the `$(document).ready(...)`) we want to hide all of them.